kubernetes/config-maps/nginx/ssl.conf

```diff
@@ -0,0 +1,26 @@
+# Disable SSLv3
+ssl_protocols              TLSv1 TLSv1.1 TLSv1.2;
+
+# Enable server-side protection
+ssl_prefer_server_ciphers  on;
+ssl_ciphers                "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+
+# Specify curve P-384
+ssl_ecdh_curve             secp384r1;
+
+# Improve ssl performence
+ssl_session_cache          shared:SSL:10m;
+ssl_session_timeout        5m;
+
+# Enable OCSP stapling
+ssl_stapling               on;
+ssl_stapling_verify        on;
+
+# Enable HSTS
+add_header Strict-Transport-Security "max-age=63072000; includeSubdomains";
```

Comment:
This `max-age` has a significant gotcha. Once we switch dev or any other cluster to HTTPS, switching back to HTTP will require clearing client's cache.

Please at least add some explanations:
```
# HSTS tells the client to only access the site over HTTPS. This is mostly relevant to browsers
# but other types clients may implement it too as long as they have some persistent storage.
# See https://www.owasp.org/index.php/HTTP_Strict_Transport_Security_Cheat_Sheet
# Note that max-age here is 63072000 seconds (2 years) which means that once you start serving
# traffic over HTTPS, you won't be able to switch back to HTTP without clearing client cache.
```

Reply:
This may cause a problem when traffic with HTTP and HTTPS are allowed.